src/avro_json_encoder.erl

```diff
@@ -151,18 +176,18 @@ do_encode_type(#avro_record_type{} = T, EnclosingNamespace) ->
   {Name, NextLevelEnclosingNs} = avro:split_type_name(T, Namespace),
   SchemaObjectFields =
     [ optional_field(namespace, ns(Namespace, EnclosingNamespace),
-                     ?NS_GLOBAL, fun encode_string/1)
-    , {type,   encode_string(?AVRO_RECORD)}
-    , {name,   encode_string(Name)}
-    , optional_field(doc,       Doc,  ?NO_DOC, fun encode_string/1)
-    , optional_field(aliases,   Aliases,   [], fun encode_aliases/1)
+                     ?NS_GLOBAL, fun encode_string/1, Opt)
+    , {name, encode_name(Name, canon_ns(Namespace, EnclosingNamespace), Opt)}
```

Comment:
All named types have a `fullname` field in the type record.
e.g. `#avro_record_type.fullname`, which should be already resolved to the correct namespace.
so there should be no need of canon_ns/2, just use the fullname field instead.

Reply:
Using fullname field now.